### PR TITLE
Tracing SDK: Do not use comparison operator

### PR DESCRIPTION
**Pull request by utezduyar**
We use comparison operator. Especially on this paragraph it has confused me. Made me think that the sampling decision is based on what IsRecording returns but in reality we are trying to define what IsRecording returns.

Fixes #

## Changes

Please provide a brief description of the changes here.

For non-trivial changes, follow the [change proposal process](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CONTRIBUTING.md#proposing-a-change) and link to the related issue(s) and/or [OTEP(s)](https://github.com/open-telemetry/oteps), update the [`CHANGELOG.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md), and also be sure to update [`spec-compliance-matrix.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md) if necessary.

Related issues #

Related [OTEP(s)](https://github.com/open-telemetry/oteps) #

### CODE REVIEW

**Review by MrAlias — COMMENTED** [review]
I do not see changing from an equality symbol to an assignment one as an improvement. This is a nonsense statement to me: "assign false to IsRecording()".

File: specification/trace/sdk.md [hunk 1]
```diff
@@ -258,10 +258,10 @@ object.
 It produces an output called `SamplingResult` which contains:
 
 * A sampling `Decision`. One of the following enum values:
-  * `DROP` - `IsRecording() == false`, span will not be recorded and all events and attributes
+  * `DROP` - `IsRecording() = false`, span will not be recorded and all events and attributes
```
Comment (arminru) [inline specification/trace/sdk.md:261]:
```suggestion
  * `DROP` - `IsRecording` will be `false`, span will not be recorded and all events and attributes
```
See https://github.com/open-telemetry/opentelemetry-specification/pull/3036#pullrequestreview-1219569084 - I think this should work. cc @MrAlias

File: specification/trace/sdk.md [hunk 2]
```diff
@@ -258,10 +258,10 @@ object.
...
   will be dropped.
-  * `RECORD_ONLY` - `IsRecording() == true`, but `Sampled` flag MUST NOT be set.
-  * `RECORD_AND_SAMPLE` - `IsRecording() == true` AND `Sampled` flag MUST be set.
+  * `RECORD_ONLY` - `IsRecording() = true`, but `Sampled` flag MUST NOT be set.
+  * `RECORD_AND_SAMPLE` - `IsRecording() = true` AND `Sampled` flag MUST be set.
```
Comment (arminru) [inline specification/trace/sdk.md:264]:
```suggestion
  * `RECORD_ONLY` - `IsRecording` will be `true`, but `Sampled` flag MUST NOT be set.
  * `RECORD_AND_SAMPLE` - `IsRecording` will be `true` and `Sampled` flag MUST be set.
```
See https://github.com/open-telemetry/opentelemetry-specification/pull/3036#pullrequestreview-1219569084 - I think this should work. cc @MrAlias

File: specification/trace/sdk.md [hunk 3]
```diff
@@ -258,10 +258,10 @@ object.
 It produces an output called `SamplingResult` which contains:
 
 * A sampling `Decision`. One of the following enum values:
-  * `DROP` - `IsRecording() == false`, span will not be recorded and all events and attributes
+  * `DROP` - `IsRecording` will be `false`, span will not be recorded and all events and attributes
   will be dropped.
-  * `RECORD_ONLY` - `IsRecording() == true`, but `Sampled` flag MUST NOT be set.
-  * `RECORD_AND_SAMPLE` - `IsRecording() == true` AND `Sampled` flag MUST be set.
+  * `RECORD_ONLY` - `IsRecording` will be `true`, but `Sampled` flag MUST NOT be set.
+  * `RECORD_AND_SAMPLE` - `IsRecording` will be `true` and `Sampled` flag MUST be set.
```
Comment (MrAlias) [inline specification/trace/sdk.md:264]:
```suggestion
  * `RECORD_ONLY` - `IsRecording` will be `true`, but the `Sampled` flag MUST NOT be set.
  * `RECORD_AND_SAMPLE` - `IsRecording` will be `true` and the `Sampled` flag MUST be set.
```

File: specification/trace/sdk.md [hunk 4]
```diff
@@ -258,10 +258,10 @@ object.
 It produces an output called `SamplingResult` which contains:
 
 * A sampling `Decision`. One of the following enum values:
-  * `DROP` - `IsRecording() == false`, span will not be recorded and all events and attributes
+  * `DROP` - `IsRecording` will be `false`, span will not be recorded and all events and attributes
```
Comment (MrAlias) [inline specification/trace/sdk.md:261]:
```suggestion
  * `DROP` - `IsRecording` will be `false`, the span will not be recorded and all events and attributes
```

**Review by MrAlias — APPROVED** [review]

**Review by arminru — APPROVED** [review]

**Pull request merged**